CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/state/DataAccessPreferenceKeys.java

```diff
@@ -109,11 +113,11 @@ public static File getPreviousDataAccessResultsDir() {
      *
      * @return The preference as a directory; null if the directory is not set.
      */
-    private static File getDir(final String pref) {
+    protected static File getDir(final String pref) {
         final Preferences prefs = NbPreferences.forModule(DataAccessPreferenceKeys.class);
         final String s = prefs.get(pref, "");
 
-        return !s.isEmpty() ? new File(s) : null;
+        return StringUtils.isNotEmpty(s) ? new File(s) : null;
```

Comment:
Null pointer was raised, so safe check for emptiness was used

Reply:
great, so the unit test found some issues and you fixed them as well. good on you @aldebaran30701